src/Sulu/Component/HttpCache/ProxyClient/Symfony.php

```diff
@@ -102,79 +103,60 @@ protected function queueRequest($method, $url, array $headers = [])
-                $request->getHeaders()
+            $promises[] = $promise = $this->client->sendAsync($request);
+            $promise->then(
+                function (ResponseInterface $res) {
+                },
+                function (RequestException $exception) use ($collection) {
+                    if ($exception instanceof ConnectException) {
+                        // Caching proxy unreachable
+                        $collection->add(
+                            ProxyUnreachableException::proxyUnreachable(
+                                $exception->getRequest()->getUri()->getHost(),
+                                $exception->getMessage(),
+                                $exception
+                            )
+                        );
+                    } elseif ($exception instanceof RequestException) {
+                        // Other error
+                        $collection->add(
+                            ProxyResponseException::proxyResponse(
+                                $exception->getRequest()->getUri()->getHost(),
+                                $exception->getCode(),
+                                $exception->getMessage(),
+                                $exception
+                            )
+                        );
+                    } else {
+                        // Unexpected exception type
+                        $collection->add($exception);
+                    }
+                }
             );
-            $allRequests[] = $proxyRequest;
-        }
-
-        try {
-            $this->client->send($allRequests);
-        } catch (MultiTransferException $e) {
-            $this->handleException($e);
-        }
-    }
-
-    /**
-     * Handle request exception.
-     *
-     * @param MultiTransferException $exceptions
-     *
-     * @throws ExceptionCollection
-     */
-    protected function handleException(MultiTransferException $exceptions)
-    {
-        $collection = new ExceptionCollection();
-
-        foreach ($exceptions as $exception) {
-            if ($exception instanceof CurlException) {
-                // Caching proxy unreachable
-                $e = ProxyUnreachableException::proxyUnreachable(
-                    $exception->getRequest()->getHost(),
-                    $exception->getMessage(),
-                    $exception
-                );
-            } elseif ($exception instanceof RequestException) {
-                // Other error
-                $e = ProxyResponseException::proxyResponse(
-                    $exception->getRequest()->getHost(),
-                    $exception->getCode(),
-                    $exception->getMessage(),
-                    $exception
-                );
-            } else {
-                // Unexpected exception type
-                $e = $exception;
-            }
-
-            $collection->add($e);
         }
 
-        throw $collection;
+        Promise\settle($promises);
```

Comment:
settle is a function in the Promise namespace. functions can be imported by `use function` but this PHP5.6 http://php.net/manual/en/language.namespaces.importing.php